services/swagger/swagger.tester.js

```diff
@@ -41,3 +65,73 @@ t.create('Invalid')
     message: 'invalid',
     color: 'red',
   })
+
+t.create('Not found')
+  .get(getURL)
+  .intercept(nock =>
+    nock(apiURL)
+      .get(apiGetURL)
+      .query(apiGetQueryParams)
+      .reply(200, {
+        schemaValidationMessages: [
+          {
+            level: 'error',
+            message: "Can't read from file https://example.com/example.json",
+          },
+        ],
+      })
+  )
+  .expectBadge({
+    label: 'swagger',
+    message: 'spec not found or unreadable',
+    color: 'red',
+  })
+
+t.create('Live valid json 2.0')
+  .get(
+    `${getURLBase}https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v2.0/json/petstore-expanded.json`
+  )
+  .expectBadge({
+    label: 'swagger',
+    message: 'valid',
+    color: 'brightgreen',
+  })
+
+t.create('Live valid yaml 3.0')
+  .get(
+    `${getURLBase}https://raw.githubusercontent.com/OAI/OpenAPI-Specification/master/examples/v3.0/petstore.yaml`
+  )
+  .expectBadge({
+    label: 'swagger',
+    message: 'valid',
+    color: 'brightgreen',
+  })
+
+t.create('Live valid with warnings')
+  .get(`${getURLBase}https://petstore3.swagger.io/api/v3/openapi.json`)
+  .expectBadge({
+    label: 'swagger',
+    message: 'valid',
+    color: 'brightgreen',
+  })
+
+// Isn't a spec, but valid json
+t.create('Live invalid')
```

Comment:
> One live test I'm not sure though: The invalid spec I just used the schema for a spec, but its not a spec as such. (but valid json)

I'm guessing this is the test you were referring to?

Reply:
Yes exactly